core/src/commonMain/kotlin/com/xebia/functional/xef/llm/LLM.kt

```diff
@@ -12,6 +12,12 @@ sealed interface LLM : AutoCloseable {
   val name
     get() = modelType.name
 
+  /**
+   * Copies this instance and uses [modelType] for [LLM.modelType].
+   * Has to return the most specific type of this instance!
+   */
+  fun copy(modelType: ModelType): LLM
```

Comment:
I don't like this, but I didn't see a better way. Every topmost subclass of LLM must return an instance of it's type so that models based on the original model with the same capabilities is spawned.